webapp/src/pages/HomePage/HomePage.tsx

```diff
@@ -1,10 +1,100 @@
+import { Card } from 'primereact/card';
+import foodPlaceHolder from '../../assets/images/pasta.jpg';
+import RatingStars from '../../components/molecules/RatingStars';
 import React from 'react';
-import './HomePage.css';
+import { Link } from 'react-router-dom';
+import './HomePage.scss';
+import { useQuery } from 'urql';
+
+// Query to get all recipes
+const getRecipes = `{
+  recipe {
+    created_at
+    user {
+      recipe {
+        prep_time_minutes
+        title
+        id
+      }
+      name
+    }
+    reviews {
+      rating
+    }
+  }}
+`;
+
+// * Not sure if this is useful but option an to get the avg rating number on the front end could be using:
+// ? const avgRating = arrayOfNumbers => arrayOfNumbers.reduce((a,b) => a + b, 0) / arrayOfNumbers.length
 
 export default function HomePage() {
+  const [result] = useQuery({
+    query: getRecipes,
+  });
+
+  const { data, fetching, error } = result;
+
+  if (fetching) return <p>Loading...</p>;
+  if (error)
+    return (
+      <p>
+        Oh no...looks like there is an issue finding that recipe.{' '}
+        {error.message}
+      </p>
+    );
+
+  // Get 3 most recent recipes:
+  const mostRecent = 3;
+  const mostRecentRecipeList = data.recipe.slice(0, mostRecent);
+
   return (
-    <div className="homepage">
-      <h1>I&apos;m a Home Page</h1>
-    </div>
+    <>
+      <div className="homepage">
+        <div className="homepage__welcome-box">
+          <h1 className="homepage__text">Welcome to GR Recipes.</h1>
+          <p className="homepage__text">
+            Where you&apos;ll find crème de la crème of people and recipes
+          </p>
+        </div>
+      </div>
+      <h2 className="homepage__header homepage__header--left "> New Recipes</h2>
+      <div className="suggested-recipe">
+        {mostRecentRecipeList.map((recipe: any) => {
```

Comment:
_Ideally_ we'd be using the Recipe type from the generated types here, if it's a 5min fix great if not don't worry about it